pyprep/find_noisy_channels.py

```diff
@@ -447,10 +448,18 @@ def find_bad_by_ransac(
             The duration (in seconds) of each RANSAC correlation window. Defaults
             to 5 seconds.
         channel_wise : bool, optional
-            Whether RANSAC should be performed one channel at a time (lower RAM
-            demands) or in chunks of as many channels as can fit into the
-            currently available RAM (faster). Defaults to ``False`` (i.e., using
-            the faster method).
+            Whether RANSAC should predict signals for whole chunks of channels
+            at once instead of predicting signals for each RANSAC window
+            individually. Channel-wise RANSAC generally has higher RAM demands
+            than window-wise RANSAC (especially if `max_chunk_size` is
+            ``None``), but can be faster on systems with lots of RAM to spare.
+            Defaults to ``False``.
```

Comment:
Okay I think I am a little confused by this.

We can run RANSAC in the following ways:

1. over all channels and over the whole time --> this is how I coded it back then
    1. needs huge amounts of RAM
    2. but if that RAM is available, it's fast
2. one channel at a time over the whole time --> this is what @yjmantilla implemented (if I recall correctly)
    1. much less RAM needed
    2. significantly slower
3. calculate RANSAC over all channels but over windows
    1. this is what MATLAB PREP does, and what autoreject does
    2. implemented by @a-hurst 
    3. Probably the best default

There would be a fourth potential option -> running single channels over windows ... so that'd be the slowest possible, but the lowest RAM ever (can probably even run on a raspberry pi :rofl: ).

Overall I am not sure how many of these options we should offer and how to document them.

Right now, I think one option should be fine: the matlab/autoreject way of doing all channels at once but over windows ... and if too little RAM is there, we just need to make the windows smaller ... right? (or would that affect the accuracy at some point?)

what are your opinions @a-hurst @yjmantilla

Reply:
I'd thought about that too (chunking-by-channel in window-wise RANSAC), but wasn't sure how to best merge the chunking code with the windowing stuff without a massive rewrite.

Right now, window-wise still takes less RAM than one-at-a-time channel-wise unless (channel count x 5 seconds) is larger than the length of the whole recording (i.e., one-channel-at-a-time only takes less RAM if a 32-channel recording is under 2.6 minutes, a 64-channel is under 5.3 mins, a 128-channel is under 10.6 mins, etc.). Thus, in most cases window-wise should be good enough on the RAM front. Still, I wouldn't be opposed to getting that down further: with enough optimization, I could probably get PyPREP running (slowly) without swapping on my old iMac G4!

Reply:
I think window-wise is better. I think remember talking bout this in another thread. In the usual use-case window-wise will usually consume less ram than the approach i took.

I think it would be nice to offer low-ram version even though it is the slowest, maybe fitting the largest chunk of channels could be done automatically like it is done currently in the channel wise version. but is not a priority. All the work @a-hurst  is doing is quite amazing, priority should be that matlab prep comparison.

So +1 making the window-wise the default

Reply:
Thanks @yjmantilla! As for the MATLAB PREP comparison, I just got the [MATLAB side of that working today](https://github.com/a-hurst/matprep_artifacts) (automated MatPREP on GitHub actions, saving relevant data as artifacts for later testing) so expect some preliminary stuff on that end shortly :)

Reply:
![image](https://user-images.githubusercontent.com/9084751/116773857-9a72f080-aa58-11eb-8768-0c54072983fc.png)


:rofl:

Reply:
please adjust this docstr according to https://github.com/sappelhoff/pyprep/pull/66/files#r624438768 for consistency :)

Reply:
lol that commit was after I'd spent a good 30-40 minutes trying to figure out why `pop_saveset` was cryptically telling me I didn't have write access to the artifacts folder.

Turns out, if you use any keyword arguments in a MATLAB function call (e.g. `'savemode', 'onefile'`), that somehow means that positional arguments beyond the first one _no longer work_ and you have to use keyword arguments for everything else (i.e. `pop_saveset(EEG, outpath, 'savemode', 'onefile');` doesn't work, even though `pop_saveset(EEG, 'filename', outpath, 'savemode', 'onefile');` does and `pop_saveset(EEG,  outpath);` does too). And I thought passing argument names as strings was bad enough...